impeller/entity/entity_pass.cc

```diff
@@ -174,231 +182,249 @@ bool EntityPass::Render(ContentContext& renderer,
+
+  if (const auto& entity = std::get_if<Entity>(&element)) {
+    element_entity = *entity;
+    if (!position.IsZero()) {
+      // If the pass image is going to be rendered with a non-zero position,
+      // apply the negative translation to entity copies before rendering them
+      // so that they'll end up rendering to the correct on-screen position.
+      element_entity.SetTransformation(
+          Matrix::MakeTranslation(Vector3(-position)) *
+          element_entity.GetTransformation());
+    }
+  }
 
-  auto context = renderer.GetContext();
+  //--------------------------------------------------------------------------
+  /// Setup subpass element.
+  ///
 
-  std::shared_ptr<CommandBuffer> command_buffer;
-  std::shared_ptr<RenderPass> pass;
-  uint32_t pass_count = 0;
+  else if (const auto& subpass_ptr =
+               std::get_if<std::unique_ptr<EntityPass>>(&element)) {
+    auto subpass = subpass_ptr->get();
 
-  auto end_pass = [&command_buffer, &pass, &context]() {
-    if (!pass->EncodeCommands(context->GetTransientsAllocator())) {
-      return false;
+    if (subpass->delegate_->CanElide()) {
+      return EntityPass::EntityResult::Skip();
     }
 
-    if (!command_buffer->SubmitCommands()) {
-      return false;
+    if (subpass->delegate_->CanCollapseIntoParentPass() &&
+        !subpass->backdrop_filter_proc_.has_value()) {
+      // Directly render into the parent target and move on.
+      if (!subpass->OnRender(renderer, pass_context.GetRenderTarget(), position,
+                             position, stencil_depth_floor)) {
+        return EntityPass::EntityResult::Failure();
+      }
+      return EntityPass::EntityResult::Skip();
     }
 
-    return true;
-  };
+    std::optional<Rect> backdrop_coverage;
+    if (subpass->backdrop_filter_proc_.has_value()) {
+      backdrop_coverage = Rect(
+          position, Size(pass_context.GetRenderTarget().GetRenderTargetSize()));
+    }
 
-  for (const auto& element : elements_) {
-    Entity element_entity;
-
-    // =========================================================================
-    // Setup entity element for rendering ======================================
-    // =========================================================================
-    if (const auto& entity = std::get_if<Entity>(&element)) {
-      element_entity = *entity;
-      if (!position.IsZero()) {
-        // If the pass image is going to be rendered with a non-zero position,
-        // apply the negative translation to entity copies before rendering them
-        // so that they'll end up rendering to the correct on-screen position.
-        element_entity.SetTransformation(
-            Matrix::MakeTranslation(Vector3(-position)) *
-            element_entity.GetTransformation());
-      }
+    std::optional<Rect> subpass_coverage =
+        GetSubpassCoverage(*subpass, backdrop_coverage);
+
+    if (!subpass_coverage.has_value()) {
+      return EntityPass::EntityResult::Skip();
```

Comment:
What if the backdrop_coverage has a value here. We shouldn't skip.

Reply:
If a backdrop filter is set and the parent pass coverage is not cancelled out entirely by the SaveLayer bounds, this branch won't get hit.

Reply:
Need to revisit this as per this comment: https://github.com/flutter/engine/pull/33887#discussion_r891789287

Reply:
Fixed this up as per other comments.